src/routes/__tests__/auth.routes.test.js

```diff
@@ -0,0 +1,322 @@
+    Object.entries(mockLOGIN_VALID_USERS).forEach(testCase => {
+      const [key, payload] = testCase;
+      describe(key, () => {
+        let user;
+
+        beforeAll(async () => {
+          user = await createUser(uuid4(), payload.email, payload.password);
+
+          await user.save();
+        });
+        afterAll(async () => {
+          await user.delete();
+        });
+        test(key, async () => {
+          const res = await request(app)
+            .post('/api/login')
+            .send(payload);
+          expect(res.statusCode).toEqual(200);
+          expect(res.body.message).toEqual('Logged in successfully!');
+        });
+      });
+    });
+  });
+
+  describe('login fails with invalid password', () => {
+    Object.entries(mockINCORRECT_LOGIN_PASSWORD).forEach(testCase => {
+      const [key, payload] = testCase;
+      describe(key, () => {
+        let user;
+
+        beforeAll(async () => {
+          user = await createUser(uuid4(), payload.email, payload.password);
+
+          await user.save();
+        });
+        afterAll(async () => {
+          await user.delete();
+        });
+
+        test(key, async () => {
+          payload.password += 'new';
+          const res = await request(app)
+            .post('/api/login')
+            .send(payload);
+          expect(res.statusCode).toEqual(401);
+          expect(res.body.message).toEqual('Invalid Email or Password');
+        });
+      });
+    });
+  });
+
+  describe('login fails with invalid data', () => {
+    Object.entries(mockLOGIN_INVALID_USERS).forEach(testCase => {
+      const [key, payload] = testCase;
+      describe(key, () => {
+        test(key, async () => {
+          const res = await request(app)
+            .post('/api/login')
+            .send(payload);
+          expect(res.statusCode).toEqual(422);
+          expect(res.body.message).toEqual(
+            'Please provide a valid username or password',
+          );
+        });
+      });
+    });
+  });
+
+  describe('login fails with not signed up user', () => {
+    Object.entries(mockLOGIN_VALID_USERS).forEach(testCase => {
+      const [key, payload] = testCase;
+      describe(key, () => {
+        beforeAll(async () => {
+          const user = await User.findOne({ email: payload.email });
+          if (user) {
+            await user.delete();
+          }
+        });
+
+        test(key, async () => {
+          const res = await request(app)
+            .post('/api/login')
+            .send(payload);
+          expect(res.statusCode).toEqual(404);
+          expect(res.body.message).toEqual('User is not signed up');
+        });
+      });
+    });
+  });
+});
+
+describe('changePassword endpoint', () => {
+  describe('user should change the old password', () => {
+    Object.entries(mockVALID_USERS).forEach(testCase => {
+      const [key, userData] = testCase;
+      const payload = {
+        email: userData.email,
+        password: uuid4(),
```

Comment:
what does this uuid do? just creating fake passwords?

Reply:
yes it generates random passwords, and I installed it with yarn not npm

Reply:
but u committed a package-lock file which only generates when u install with npm hmm. u gotta remove that file

Reply:
![Screenshot 2021-05-02 at 20 13 01](https://user-images.githubusercontent.com/49414052/116823090-d7d39d00-ab82-11eb-9302-7f90e4694461.png)
I cannot find anything related to uuid in package-lock file

Reply:
u should remove the whole file. we don't use package-lock.

Reply:
I removed it from github and it created a delete commit for me